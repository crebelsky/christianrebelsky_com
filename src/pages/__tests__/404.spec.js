import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery } from 'gatsby';
import NotFound from '../404';

describe('NotFound', () => {
  beforeAll(() => {
    useStaticQuery.mockImplementation(() => ({
      site: {
        siteMetadata: {
          title: 'Test Site',
          description: 'Test Description',
          author: 'Test Author',
        },
      },
    }));
  });

  it('renders correctly', () => {
    const tree = renderer.create(<NotFound />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
