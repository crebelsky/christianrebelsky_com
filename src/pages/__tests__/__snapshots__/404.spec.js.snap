// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders correctly 1`] = `
<div
  className="page"
>
  <main>
    <h1>
      NOT FOUND
    </h1>
    <p>
      You just hit a route that doesn't exist... the sadness.
    </p>
  </main>
  <footer>
    <nav
      data-testid="footer-nav"
    >
      <ul>
        <li>
          <a
            href="/imprint/"
          >
            Imprint
          </a>
        </li>
        <li>
          <a
            href="/data-privacy/"
          >
            Data Privacy
          </a>
        </li>
      </ul>
    </nav>
  </footer>
</div>
`;
